app/models/container_image.rb

```diff
@@ -11,6 +11,7 @@ class ContainerImage < ApplicationRecord
   DOCKER_PULLABLE_PREFIX = "docker-pullable://".freeze
   DOCKER_PREFIXES = [DOCKER_IMAGE_PREFIX, DOCKER_PULLABLE_PREFIX].freeze
 
+  belongs_to :parent, :polymorphic => true
```

Comment:
If so, then `parent` might be a bad name choice as it conflicts with the Relationship code, should that be used in the future.